libs/model/src/comment/UpdateComment.command.ts

```diff
@@ -0,0 +1,96 @@
+import { type Command } from '@hicommonwealth/core';
+import * as schemas from '@hicommonwealth/schemas';
+import { models } from '../database';
+import { isAuthorized, type AuthContext } from '../middleware';
+import { mustExist } from '../middleware/guards';
+import {
+  emitMentions,
+  findMentionDiff,
+  parseUserMentions,
+  quillToPlain,
+  sanitizeQuillText,
+  uniqueMentions,
+} from '../utils';
+
+export function UpdateComment(): Command<
+  typeof schemas.UpdateComment,
+  AuthContext
+> {
+  return {
+    ...schemas.UpdateComment,
+    auth: [isAuthorized({})],
+    body: async ({ actor, payload }) => {
+      const { comment_id, discord_meta } = payload;
+
+      const comment = await models.Comment.findOne({
+        where: comment_id ? { id: comment_id } : { discord_meta },
+        include: [{ model: models.Thread, required: true }],
+      });
+      mustExist('Comment', comment);
```

Comment:
`mustExist` here could lead to 500 errors when the comment has been deleted no?

Reply:
yes, this is to guard against an invalid state in the system
... can also be designed as an invalid input (400), assuming the user must know something about the state, guess we can debate which pattern works better

Reply:
If the comment doesn't exist when the user attempts to update it then that's a 400 hundred error. Otherwise, I could call this route with any comment_id (even one that doesn't exist) and we would get a bunch of 500 errors reported to Rollbar.

Reply:
right... considering our Rollbar/Slack strategy, a 400 approach is desired

Reply:
I'll update this in #9150 with other small fix in the discord integration